### PR TITLE
Fix sequencer tx production

### PR DESCRIPTION
**Pull request by otherview**
### Why this change is needed

- https://github.com/obscuronet/obscuro-internal/issues/1415

### What changes were made as part of this PR

Please provide a high level list of the changes made

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/obscuronet/obscuro-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: go/enclave/enclave.go [hunk 1]
```diff
@@ -290,19 +297,9 @@ func (e *enclaveImpl) SubmitTx(tx common.EncryptedTx) (common.EncryptedResponseS
...
-	// Only the sequencer needs to maintain a transaction mempool. Other node types can return early.
-	if e.config.NodeType != common.Sequencer {
-		return encryptedResult, nil
-	}
-
-	if err = e.mempool.AddMempoolTx(decryptedTx); err != nil {
-		return nil, err
-	}
-
 	if e.config.SpeculativeExecution {
```
Comment (BedrockSquirrel) [inline go/enclave/enclave.go:300]: do you know if speculative execution is ever enabled? This changes behaviour for validator nodes if speculativeExecution is set but don't know if there's any impact (I thought that flag was for POBI and so it's probably deprecated)

**Review by BedrockSquirrel — COMMENTED** [review]
LGTM but just wanted to check if you'd looked into the txCh thing because this might cause that to start getting populated on validators.

**Review by BedrockSquirrel — APPROVED** [review]
LGTM

File: go/enclave/enclave.go (hunk 1 above)
Comment (otherview) [reply go/enclave/enclave.go:300]: ah! I literally thought the same thing. But moved it to another PR https://github.com/obscuronet/go-obscuro/pull/1092 :)

**Pull request merged**